### DISCUSSION TITLE
How to make a macOS Chromium remote index.

### DISCUSSION THREAD

**Discussion by Tackoil** [clangd/clangd#2131] [Q&A]
Hi all.

I'm trying to make a macOS build Chromium remote index on a powerful machine for avoiding the background indexing on a device without fans (MacBook Air, I mean). I run the following commands refering [this repo](https://github.com/clangd/chrome-remote-index/blob/3dd568b68ba6c2be3b06bacaa254e40de8ff955c/indexer/run.sh#L69).  But it get a mount of errors and `clangd-index-server` will get a 0 bytes index. 

```bash
clangd-indexer --executor=all-TUs compile_commands.json > proj.idx
```

But the index in the code editor with LSP (like Zed) is working quiet well. (even without the `-ferror-limit` in .clangd).
So it there some experience about this topic? How could I make a next step to figure out this problem?

**Comment by HighCommander4** [comment]
I don't have personal experience with it. But if you share the output of the command (the one with the errors), maybe it will contain some clues.

**Comment by Tackoil** [comment]
There are too much errors. I run the indexer for few seconds and paste at following. Duplicate errors has been removed.

```
error: unknown warning option '-Wno-thread-safety-reference-return'; did you mean '-Wno-thread-safety-reference'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wno-c++11-narrowing-const-reference'; did you mean '-Wno-uninitialized-const-reference'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wno-missing-template-arg-list-after-template-kw'; did you mean '-Wno-gnu-string-literal-operator-template'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wno-dangling-assignment-gsl'; did you mean '-Wno-dangling-gsl'? [-Werror,-Wunknown-warning-option]
```

**Reply by HighCommander4** [reply]
You'll need to fix the errors.

Maybe you're using an older clangd than your build compiler? Looking at the first error, `-Wno-thread-safety-reference-return` was first introduced in llvm-18.

**Reply by Tackoil** [reply]
I just checked the llvm version used by Chromium and found out it's 19. 

```txt
clang version 19.0.0git (https://chromium.googlesource.com/a/external/github.com/llvm/llvm-project ecea8371ff03c15fb3dc27ee4108b98335fd2d63)
Target: arm64-apple-darwin24.0.0
Thread model: posix
InstalledDir: /<path to chromium>/src/third_party/llvm-build/Release+Asserts/bin
```

I update the indexer and get less error than before. 

BTW, I'm not familiar with clangd. But how the code editors get the correctly clangd version? As the fact, the clangd version in system is 16 at now. And Zed just open it.

```txt
Apple clangd version 16.0.0 (clang-1600.0.22.7)
Features: mac+xpc
Platform: arm64-apple-darwin24.0.0
```

<img width="665" alt="image" src="https://github.com/user-attachments/assets/3855cc05-c4c8-4463-abfd-547df731b05b">
